powerstore/resource_volume.go

```diff
@@ -298,10 +341,19 @@ func (r resourceVolume) Create(ctx context.Context, req tfsdk.CreateResourceRequ
 		)
 		return
 	}
+	// Get Volume Group Mapping details from API
+	volGroupMapping, err := r.p.client.PStoreClient.GetVolumeGroupsByVolumeID(context.Background(), volCreateResponse.ID)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error fetching volume host mapping",
+			"Could not create volume, unexpected error: "+err.Error(),
+		)
+		return
+	}
 	log.Printf("After Volume create call")
 
 	result := models.Volume{}
-	updateVolState(&result, volResponse, hostMapping, &plan, "Create")
+	updateVolState(&result, volResponse, hostMapping, volGroupMapping, &plan, "Create")
```

Comment:
try to use https://github.com/dell/terraform-provider-powerstore/blob/dev/powerstore/defaults.go#L14